python/tank/pipelineconfig_factory.py

```diff
@@ -89,12 +88,11 @@ def _from_entity(entity_type, entity_id, force_reread_shotgun_cache):
         
         if config_context_path not in local_pc_paths:
             # the tank command / api proxy which this session was launched for is *not*
-            # associated with the given entity type and entity id!            
-            raise TankError("Error launching %s %s from the configuration located in '%s'. "
-                            "This config is not associated with that project. For a list of "
-                            "which configurations can be used with this project, go to the "
-                            "Pipeline Configurations page in Shotgun "
-                            "for the project." % (entity_type, entity_id, config_context_path))
+            # associated with the given entity type and entity id!
+            raise TankError("The Toolkit configuration in '%s' is not associated with "
+                            "%s %s. To see which Toolkit configurations are available for a "
+                            "Project, go to the Pipeline Configurations page "
+                            "in Shotgun." % (config_context_path, entity_type, entity_id))                        
```

Comment:
I'm a little bothered by the different terminology and nomenclature we use in various parts of Toolkit. "config", "Toolkit configuration", "pipeline configuration", "project config"... they all mean the same thing and we use them interchangeably internally. But that's confusing for users. Perhaps this isn't the right place to start this conversation but we should huddle and agree upon some standards. I guess ideally we are explicit and I'd use "pipeline configuration" since that's also what it's called in Shotgun so it's consistent. 

The issue here is that the project for the entity and the project for the pipeline configuration don't match. I think we lost a little bit of the Project element of this issue with this change that we could add back

And I'm not sure we need to also tell them it's a Toolkit thing. 

> The pipeline configuration in '/some/path/to/config' is associated with a different Project from Shot 123. To see which pipeline configurations are available for a Project open the Pipeline Configurations page in Shotgun. 

I'm not totally satisfied with that but I *think* it's better :whale: